pyro/infer/mcmc/hmc.py

```diff
@@ -187,7 +191,8 @@ def compiled(*zi):
             return potential_energy
 
         with pyro.validation_enabled(False), optional(ignore_jit_warnings(), self._ignore_jit_warnings):
-            self._compiled_potential_fn = torch.jit.trace(compiled, vals, check_trace=False)
+            self._compiled_potential_fn = torch.jit.trace(compiled, vals, check_trace=False,
+                                                          optimize=self._jit_optimize)
```

Comment:
@neerajprad Do you think it would be more useful to add a `jit_options` dict and call
```py
torch.jit.trace(compiled, vals, check_trace=False, **jit_options)
```
This costs an extra level of indirection (`HMC(..., jit_options={'optimize': False})`) but will avoid duplication of possibly multiple future jit options in the `HMC` class.

Reply:
+1. It will be much better to consolidate all JIT options into a `jit_options` dict!